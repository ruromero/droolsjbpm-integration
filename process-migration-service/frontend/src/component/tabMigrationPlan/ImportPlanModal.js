import React from "react";
import PropTypes from "prop-types";

import { Button, Modal } from "patternfly-react";
import Form from "patternfly-react/dist/js/components/Form/Form";
import {
  FormGroup,
  FormControl
} from "patternfly-react/dist/js/components/Form";
import ControlLabel from "patternfly-react/dist/js/components/Form/ControlLabel";
import Notification from "../Notification";
import { ALERT_TYPE_ERROR } from "patternfly-react/dist/js/components/Alert/AlertConstants";

export default class ImportPlanModal extends React.Component {
  constructor() {
    super();
    this.state = { showModal: false, isValid: false, errorMsg: "" };
  }
  onCloseModal = () => {
    this.setState({ showModal: false });
  };
  onOpenModal = () => {
    this.setState({
      showModal: true,
      planObject: {},
      plan: ""
    });
  };
  onChange = event => {
    let planObject;
    try {
      planObject = JSON.parse(event.target.value);
    } catch (e) {
      planObject = {};
    }
    this.setState({
      isValid: planObject !== {},
      planObject,
      plan: event.target.value
    });
  };

  submit = () => {
    this.props
      .onImport(this.state.planObject)
      .then(() => this.onCloseModal())
      .catch(() => this.setState({ errorMsg: "Unable to import the plan" }));
  };

  render() {
    const notification = (
      <Notification
        type={ALERT_TYPE_ERROR}
        message={this.state.errorMsg}
        onDismiss={() => this.setState({ errorMsg: "" })}
      />
    );
    const formBody = (
      <Form>
        {this.state.errorMsg && notification}
        <FormGroup controlId="importText">
          <ControlLabel>Migration plan</ControlLabel>
          <FormControl
            componentClass="textarea"
            onChange={this.onChange}
            value={this.state.plan}
            rows={10}
          />
        </FormGroup>
      </Form>
    );

    return (
      <React.Fragment>
        <Button bsStyle="default" onClick={this.onOpenModal}>
          Import
        </Button>

        <Modal show={this.state.showModal} onHide={this.onCloseModal} size="lg">
          <Modal.Header>
            <Modal.CloseButton onClick={this.onCloseModal} />
            <Modal.Title>Import migration plan</Modal.Title>
          </Modal.Header>
          <Modal.Body>{formBody}</Modal.Body>
          <Modal.Footer>
            <Button
              bsStyle="default"
              className="btn-cancel"
              onClick={this.onCloseModal}
            >
              Cancel
            </Button>
            <Button
              bsStyle="primary"
              onClick={this.submit}
              disabled={!this.state.isValid}
            >
              Import
            </Button>
          </Modal.Footer>
        </Modal>
      </React.Fragment>
    );
  }
}

ImportPlanModal.propTypes = {
  onImport: PropTypes.func.isRequired
};
