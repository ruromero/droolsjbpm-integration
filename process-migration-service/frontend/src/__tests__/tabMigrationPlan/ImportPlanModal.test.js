import renderer from "react-test-renderer";
import React from "react";
import ImportPlanModal from "../../component/tabMigrationPlan/ImportPlanModal";

test("ImportPlanModal renders correctly using snapshot", () => {
  const mockFn = jest.fn();
  const tree = renderer.create(<ImportPlanModal onImport={mockFn} />).toJSON();
  expect(tree).toMatchSnapshot();
});
