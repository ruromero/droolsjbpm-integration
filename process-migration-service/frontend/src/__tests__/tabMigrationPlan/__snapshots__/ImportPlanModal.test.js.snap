// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportPlanModal renders correctly using snapshot 1`] = `
<button
  className="btn btn-default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Import
</button>
`;
